src/slashCommand/commands/randomcabin.command.ts

```diff
@@ -14,35 +19,144 @@ export default class RandomCabinCommand implements BaseCommand {
       .setName(this.name)
-      .setDescription('Finds a random cabin available at your dates')
+      .setDescription(
+        'Get a random cabin. Optional: Provide dates to make sure it is available.',
+      )
       .setDMPermission(true)
       .addStringOption((option) =>
         option
           .setName('check-in')
           .setDescription('When do you want to arrive (yyyy-mm-dd)?')
-          .setRequired(true),
+          .setRequired(false),
       )
       .addStringOption((option) =>
         option
           .setName('check-out')
           .setDescription('When do you want to leave (yyyy-mm-dd)?')
-          .setRequired(true),
+          .setRequired(false),
       );
   }
 
-  // Defer response
-  // https://discordjs.guide/interactions/slash-commands.html#editing-responses
-
   public async handleCommand(
     interaction: ChatInputCommandInteraction<CacheType>,
   ): Promise<void> {
-    const checkIn = interaction.options.getString('check-in', true);
-    const checkOut = interaction.options.getString('check-out', true);
-    await interaction.reply('Not quite there yet, but working on it!');
+    await interaction.deferReply({ ephemeral: true });
+    const checkIn = interaction.options.getString('check-in', false);
+    const checkOut = interaction.options.getString('check-out', false);
+
+    if (checkIn === null || checkOut === null) {
+      await this.handleRandomCabinWithoutDates(interaction);
+    } else {
+      await this.handleRandomCabinWithDates(interaction, checkIn, checkOut);
+    }
+  }
+
+  private async handleRandomCabinWithoutDates(
+    interaction: ChatInputCommandInteraction<CacheType>,
+  ): Promise<void> {
+    const cabin = await this.cabinService.getRandomCabin();
+
+    if (cabin === null) {
+      await interaction.editReply(
+        `I'm sorry, but looks like I could not find any cabin for you. :crying_cat_face:`,
+      );
+    } else {
+      this.editReplyWithCabin(interaction, cabin);
+    }
+  }
+
+  private async handleRandomCabinWithDates(
+    interaction: ChatInputCommandInteraction<CacheType>,
+    checkIn: string | null,
+    checkOut: string | null,
+  ): Promise<void> {
+    const bookingDates = await new BookingDatesSchema().validate(
+      checkIn,
+      checkOut,
+    );
+
+    if (bookingDates === undefined) {
+      await interaction.editReply(
+        `Looks like there's something wrong with your dates.`,
+      );
+      return;
+    }
+
+    const cabin = await this.cabinService.getRandomAvailableCabin(bookingDates);
+
+    if (cabin === null) {
+      await interaction.editReply(
+        `I'm sorry, but looks like I could not find any cabin for you. :crying_cat_face:`,
+      );
+    } else {
+      this.editReplyWithCabin(interaction, cabin, bookingDates);
+    }
+  }
+
+  // TODO can we reuse the method from bot service or write one that can be used by both modules? Maybe in cabin service or cabin database service?
```

Comment:
Building the embed is a presentational layer concern so I wouldn't mix it with database layers or business services.

Something from the bot module is a good idea!